torch/csrc/jit/ir.cpp

```diff
@@ -771,9 +775,10 @@ bool Node::matches(
 }
 
 bool Node::mustBeNone() const {
-  return kind_ == prim::Constant && !this->hasAttributes() &&
-      (output()->type()->cast<OptionalType>() ||
-       output()->type() == NoneType::get());
+  return kind_ == prim::AutogradZero ||
```

Comment:
IIRC the only place where `AutogradZero` and `AutogradAdd` is added is when we specialize the inputs, and we don't need to handle this case in `unwrapoptional` related stuff if we don't specialize `Optional[Tensor]` to `AutogradZero` type (as that one need more discussions). but it might be ok for us to add a pre-check here. cc @zdevito

Reply:
i think it would be safer & more future-proof to add the check